ArduCopter/system.cpp

```diff
@@ -13,6 +13,22 @@ static void failsafe_check_static()
     copter.failsafe_check();
 }
 
+static bool validate_fence_frame(float newframe) 
+{    
+    if ((int)newframe > 3 || (int)newframe < 0) {
+        return false;
+    } else {
+        return AP::fence()->conv_max_alt_frame((int)newframe);
```

Comment:
These validation routines were never meant to modify values - only validate them.

Coupling parmeters like this is going to lead to all sorts of confusion.  Pretty sure we want to leave it up to the user to set both the frame and altitude at the same time.